lib/mongoid/extensions/big_decimal.rb

```diff
@@ -58,7 +68,31 @@ def demongoize(object)
         #
         # @return [ String, nil ] A String representing the object or nil.
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              mongoize_to_decimal128(object)
+            elsif object.numeric?
+              object.to_s
+            end
+          end
+        end
+
+        private
+
+        def demongoize_from_decimal128(object)
+          if Mongoid.map_big_decimal_to_decimal128
+            object.to_big_decimal
+          else
+            raise Mongoid::Errors::UnmappedBSONType.new(object)
+          end
+        end
+
+        def mongoize_to_decimal128(object)
+          if object.is_a?(BigDecimal)
+            BSON::Decimal128.new(object)
+          elsif object.numeric?
+            BSON::Decimal128.new(object.to_s)
+          end
```

Comment:
I think we need an else condition here?

Reply:
From the tests, it actually seems like returning `nil` is intended here. Does that make sense?

Reply:
What is the purpose of this method? If it is only called from mongoize above why not put all logic there?

Reply:
yup it seems like they abstracted some of the logic into its own function, even though it seems like it's only called once

Reply:
I removed those functions and put the logic back into the bigger functions